comments_blackbox_test.go

```diff
@@ -0,0 +1,190 @@
+type CommentsSuite struct {
+	gormsupport.DBTestSuite
+	db    *gormapplication.GormDB
+	clean func()
+}
+
+func (s *CommentsSuite) SetupTest() {
+	s.db = gormapplication.NewGormDB(s.DB)
+	s.clean = cleaner.DeleteCreatedEntities(s.DB)
+}
+
+func (s *CommentsSuite) TearDownTest() {
+	s.clean()
+}
+
+func (s *CommentsSuite) unsecuredController() (*goa.Service, *CommentsController) {
+	svc := goa.New("Comments-service-test")
+	commentsCtrl := NewCommentsController(svc, s.db)
+	return svc, commentsCtrl
+}
+
+func (s *CommentsSuite) securedControllers(identity account.Identity) (*goa.Service, *WorkitemController, *WorkItemCommentsController, *CommentsController) {
+	pub, _ := almtoken.ParsePublicKey([]byte(almtoken.RSAPublicKey))
+	priv, _ := almtoken.ParsePrivateKey([]byte(almtoken.RSAPrivateKey))
+	svc := testsupport.ServiceAsUser("Comment-Service", almtoken.NewManager(pub, priv), identity)
+	workitemCtrl := NewWorkitemController(svc, s.db)
+	workitemCommentsCtrl := NewWorkItemCommentsController(svc, s.db)
+	commentsCtrl := NewCommentsController(svc, s.db)
+	return svc, workitemCtrl, workitemCommentsCtrl, commentsCtrl
+}
+
+// createWorkItem creates a workitem that will be used to perform the comment operations during the tests.
+func (s *CommentsSuite) createWorkItem(identity account.Identity) string {
+	createWorkitemPayload := app.CreateWorkitemPayload{
+		Data: &app.WorkItem2{
+			Type: APIStringTypeWorkItem,
+			Attributes: map[string]interface{}{
+				workitem.SystemTitle: "Title",
+				workitem.SystemState: workitem.SystemStateNew},
+			Relationships: &app.WorkItemRelationships{
+				BaseType: &app.RelationBaseType{
+					Data: &app.BaseTypeData{
+						Type: "workitemtypes",
+						ID:   workitem.SystemBug,
+					},
+				},
+			},
+		},
+	}
+	userSvc, workitemCtrl, _, _ := s.securedControllers(identity)
+	_, wi := test.CreateWorkitemCreated(s.T(), userSvc.Context, userSvc, workitemCtrl, &createWorkitemPayload)
+	workitemId := *wi.Data.ID
+	s.T().Log(fmt.Sprintf("Created workitem with id %v", workitemId))
+	return workitemId
+}
+
+// createWorkItemComment creates a workitem comment that will be used to perform the comment operations during the tests.
+func (s *CommentsSuite) createWorkItemComment(identity account.Identity, workitemId string) uuid.UUID {
+	createWorkItemCommentPayload := app.CreateWorkItemCommentsPayload{
+		Data: &app.CreateComment{
+			Type: "comments",
+			Attributes: &app.CreateCommentAttributes{
+				Body: "a comment",
+			},
+		},
+	}
+	userSvc, _, workitemCommentsCtrl, _ := s.securedControllers(identity)
+	_, comment := test.CreateWorkItemCommentsOK(s.T(), userSvc.Context, userSvc, workitemCommentsCtrl, workitemId,
+		&createWorkItemCommentPayload)
+	commentId := *comment.Data.ID
+	s.T().Log(fmt.Sprintf("Created comment with id %v", commentId))
+	return commentId
+}
+
+func (s *CommentsSuite) TestShowCommentWithoutAuth() {
```

Comment:
also, I like the idea of having this piece of code declared in a single place. This reduces the amount of code in the tests themselves, and furthermore, it does not add any value to the test itself. But this can be discussed ;-)